plugins/kotlin-serialization/kotlin-serialization-compiler/src/org/jetbrains/kotlinx/serialization/compiler/resolve/KSerializerDescriptorResolver.kt

```diff
@@ -366,13 +366,20 @@ object KSerializerDescriptorResolver {
 
         val args = mutableListOf<ValueParameterDescriptor>()
         var i = 0
+
+        val targ = TypeParameterDescriptorImpl.createWithDefaultBound(f, Annotations.EMPTY, false, Variance.INVARIANT, Name.identifier("T"), 0)
+        val typeArguments = listOf(TypeProjectionImpl(targ.defaultType))
+
+        val koutputClass = KotlinTypeFactory.simpleNotNullType(Annotations.EMPTY,
```

Comment:
`KOutput` does not have type parameters, so I have no idea why it helps here

Reply:
Probably by accident. I didn't look carefully at koutput